src/query.rs

```diff
@@ -1,304 +1,228 @@
-use crate::world::{Entity, World, WorldQuery};
+use parking_lot::{MappedRwLockReadGuard, MappedRwLockWriteGuard};
+
+use crate::{
+    sparse_set::SparseSet,
+    world::{Entity, World, WorldQuery},
+};
 
 pub trait Query<'a>: Sized {
-    type Short<'b>;
+    type Short<'b, 'c, 'd, 'e, 'f>;
 
     #[track_caller]
-    fn get_components(world: &'a World, f: impl for<'b> FnMut(Entity, Self::Short<'b>));
+    fn get_components(
+        world: &'a World,
+        f: impl for<'b, 'c, 'd, 'e, 'f> FnMut(Entity, Self::Short<'b, 'c, 'd, 'e, 'f>),
```

Comment:
This is a bit ridiculous but it works. One lifetime per maximum number of components.

I can probably avoid that by some static lifetime bound somewhere, but it want obvious where